nitro-protocol/contracts/LedgerFinancingApp.sol

```diff
@@ -0,0 +1,151 @@
+pragma solidity 0.8.17;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+// LedgerFinancingApp is a ForceMoveApp that allows a intermediary to earn interest
+// on a deposit. It fuctions as a ConsensusApp with the following additional rule:
+//   - the intermediary can unilaterally transition from state n to state n+1,
+//     forcing calculated interest into the intermediary's Outcome allocation
+contract LedgerFinancingApp is IForceMoveApp {
+    struct Funds {
+        address[] asset;
+        uint256[] amount;
+    }
+
+    struct InterestAppData {
+        // a per-day simple interest rate (daily percentage yield), as a fraction.
+        // ie, 1% per day is represented with dpyNum = 1 and dpyDen = 100.
+        //
+        // Lower numbers (eg, fraction in simplest terms) produce least risk of overflow.
+        uint128 dpyNum;
+        uint128 dpyDen;
+        // the block number of the latest principal adjustment
+        uint256 blocknumber;
+        // the current principal. Decreases as the serviceProvider earns via the channel.
+        Funds principal;
+        // the total interest collected so far. Strictly increasing.
+        // The value of the intermediary's allocation can never be less than this:
+        // ie, when the intermediary's collectedInterest grows to be equal to their
+        // allocation, the channel is effectively spent and can be concluded.
+        Funds collectedInterest;
+    }
+
+    enum AllocationIndicies {
+        intermediary, // financier: makes initial deposit and earns interest
+        serviceProvider // borrower: recovers service fees from intermediary's deposit
+    }
+
+    function requireStateSupported(
+        FixedPart calldata fixedPart,
+        RecoveredVariablePart[] calldata proof,
+        RecoveredVariablePart calldata candidate
+    ) external view override {
+        if (proof.length == 0) {
+            // unanimous consensus check
+            require(
+                NitroUtils.getClaimedSignersNum(candidate.signedBy) ==
+                    fixedPart.participants.length,
+                '!unanimous; |proof|=0'
+            );
+            return;
+        } else if (proof.length == 1) {
+            // check that proof[0] -> candidate respects the stated interest rate.
+            // Requires:
+            //  - proof state is unanimous
+            //  - candidate state immediately follows proof state (by turnNum)
+            //  - the intermediary has not taken more funds than owed according
+            //    to the interest rate agreement of the channel
+            require(
+                NitroUtils.getClaimedSignersNum(proof[0].signedBy) == 2,
```

Comment:
L51 was originally also a hardcoded 2, but the linter blocked merging because `fixedPart` was unused in `requireStateSupported`.

Reply:
OK I think you can get around that by removing the variable name from the function signature.

Reply:
I had forgotten (or never knew) that unnamed function parameters were a thing in solidity. That's better here because it decreases the mental load for understanding the function.

4a847b77